### ISSUE TITLE
False alarm on closure with anonymous method

### ISSUE THREAD

**Issue by Maxwe11** [microsoft/RoslynClrHeapAllocationAnalyzer#18]
```
var ints = new[] { 5, 4, 3, 2, 1 };

Array.Sort(ints, delegate(int x, int y) { return x - y; }); // False alarm

Array.Sort(ints, (x, y) => x - y); // Ok!
```

**Comment by mjsabby** [comment]
Yup, pull request welcome. Otherwise I'll get to it soon.

**Comment by cezarypiatek** [comment]
This one is working fine

```
using System;
public class MyClass {
    public void Testing() {
        var ints = new[] { 5, 4, 3, 2, 1 };
        Array.Sort(ints, delegate(int x, int y) { return x - y; }); // False alarm
        Array.Sort(ints, (x, y) => x - y); // Ok!
    }
}
```

However, when we use struct  instead of class this will result with 2 diagnostics

```csharp
```

**Comment by cezarypiatek** [comment]
@mjsabby Should we report this one-time allocations or not? If not, this issue should be closed, because the original problem has been solved by #79
